### PR TITLE
docs: add Docusaurus site

### LINKED ISSUE #5214: Dead link: https://react-bootstrap.github.io/components/popovers/
Are there any docs for popover? 

https://react-bootstrap.github.io/components/popovers/

The link is dead on the docs website.

### LINKED ISSUE #5241: A little gap between page and navbar for chrome.
Screen Shot 2020-06-16 at 16 39 29

### LINKED ISSUE #5310: Convert all .js to Typescript in www (the frontend docs)
It will great to have all the files in www/src in TypeScript

### LINKED ISSUE #5317: Add navigation option to next page.
## Is your feature request related to a problem? Please describe

While going through documentation , it is not user friendly to always go side-navigation bar for the next page.
## Describe the solution you'd like

At the end of every page add a button linked to the next page as provided [here](https://reactjs.org/docs/components-and-props.html).

### LINKED ISSUE #6277: feat: Scrollspy on Table of contents.
### Is your feature request related to a problem? Please describe

As of now, it is little difficult to find the position of page according to the TOC.

### Describe the solution you'd like

Adding scrollspy will help the user to find the current position of the page as per the table of content.

### Describe alternatives you've considered

_No response_

### Additional context

I would like to work on it.

### LINKED ISSUE #6510: Proposal: Switch to docusaurus for docs site
It's better suited for this sort of thing, allows versioning and i18n, and supports the same sort of codeblocks, and react-metadata (b/c of plugins i wrote).

Downsides are that it's a bit harder to get a lot of these benefits if we want to build our own theme using react-bootstrap (It's generally my position that UI libraries should dogfood their own things).

### PR DESCRIPTION
**Pull request by kyletsang**
Docusaurus site

Preview: https://deploy-preview-6553--react-bootstrap.netlify.app/

Checklist:
- [x] Ads
- [x] Port content from old site
- [x] Algolia integration 
- [x] Style everything.  I've included the bootstrap styles globally so it messed a lot of things up
- [x] Fix some code examples
- [x] Some content could use some review
- [x] Link local `react-bootstrap` library.  Currently uses the npm release.
- [x] Generate API docs from source
- [x] Fix jsdoc annotations in props table 
- [x] Rename www2 folder to www and delete Gatsby site
- [x] Setup deployment
- [ ] ???

Closes #6510 
Closes #6277
Closes #5317
Closes #5310
Closes #5241 
Closes #5214

### CODE REVIEW

**Comment by jquense** [comment]
O man this is awesome. Really nice to see all the bespoke Gatsby stuff go away

**Comment by kyletsang** [comment]
@jquense, I think I got everything done for the site except for 1 thing.

I'm having trouble getting the site to link to the local RB source files properly.  The actual linking works (I can import the components), but during render, the site is throwing an error about "invalid hook call" as if I had multiple versions of React installed.  Not sure why I'm getting this error.

Are you available to help with this?

**Comment by kyletsang** [comment]
@jquense @golota60, this is ready for review

**Review by golota60 — COMMENTED** [review]
looks awesome - couple comments. I didn't focus on code that much, mostly I've just checked out the branch and tried everything out, so:
- [subjective] `--ifm-navbar-background-color` looks pretty strange to me - this is just my opinion so feel free to ignore - personally i'd go with something with less contrast - maybe more dark-blue(e.g. `#003870`)? @jquense probably has an opinion about it :smile:  
- [`Getting started`](http://localhost:3000/docs/category/getting-started)(and each subsequent "container" page, eg. `Layout`, `Forms` etc.) page - everything appears underlined for me
- On each page at the bottom there are link buttons to prev/next page - those appear underlined, I don't think they should be(?)
- [`Layout` -> `Grid`](http://localhost:3000/docs/layout/grid) - first few examples are not really showing much anymore, cause the mage is narrower
- [`Forms` -> `Input Group` -> `Buttons with dropdowns`](http://localhost:3000/docs/forms/input-group#buttons-with-dropdowns) -> `___DEV__ is not defined`  also `Segmented buttons` part + API part seems to be broken too
- [`Components` "wrapper" page](http://localhost:3000/docs/category/components) has a placeholder copy - I'd remove it/improve it
- [`Breadcrumbs` example](http://localhost:3000/docs/components/breadcrumb#example) "home" appears like it's got a slight css problem
- [`Components` -> `Button group` -> `Nesting`/`Vertical variation`](http://localhost:3000/docs/components/button-group#nesting) -> `__DEV__ is not defined`
- [`Components` -> `Dropdowns`](http://localhost:3000/docs/components/dropdowns) -> `__DEV__ is not defined` on every example
- [`Components` -> `Navbars` -> `Overview`/`Scrolling`/`Responsive behaviors`/`Offcanvas` ](http://localhost:3000/docs/components/navbar) -> `__DEV__ is not defined` and no icon in `Brand` section(image doesn't load)
- [`Components` -> `Navs and tabs` -> `Using dropdowns`](http://localhost:3000/docs/components/navs#using-dropdowns) `__DEV__ is not defined` and in API section `onSelect` appears to be a code block, which looks pretty weird
- [`Components` -> `Pagination` -> `API`](http://localhost:3000/docs/components/pagination#pagination) seems to slightly broken

^all the hyperlinks point to localhost


But overall I love how all of this looks - this is a big improvement over gatsby both from usage and visual standpoint

**Comment by kyletsang** [comment]
Thanks for the detailed review!  I should've checked the examples after fixing the local components integration 🤦‍♂️

**Comment by kyletsang** [comment]
Fixed all the issues except these:

> [subjective] --ifm-navbar-background-color looks pretty strange to me - this is just my opinion so feel free to ignore - personally i'd go with something with less contrast - maybe more dark-blue(e.g. #003870)? @jquense probably has an opinion about it 😄

Yeah not sure what to do with this.  I'd like to try implementing the dark mode on the site when we integrate BS 5.3, so we could probably change more.  Future PR?

> [Layout -> Grid](http://localhost:3000/docs/layout/grid) - first few examples are not really showing much anymore, cause the mage is narrower

These examples were the same as the old site and you can kinda see the difference when you resize the screen

File: src/Pagination.tsx [hunk 1]
```diff
@@ -28,14 +28,6 @@ const propTypes = {
...
 };
 
-/**
- * @property {PageItem} Item
- * @property {PageItem} First
- * @property {PageItem} Prev
- * @property {PageItem} Ellipsis
- * @property {PageItem} Next
- * @property {PageItem} Last
- */
```
Comment (golota60) [inline src/Pagination.tsx:38]: not sure why those are removed

File: www/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+$ USE_SSH=true yarn deploy
+```
+
+Not using SSH:
+
+```
+$ GIT_USER=<Your GitHub username> yarn deploy
+```
+
+If you are using GitHub pages for hosting, this command is a convenient way to build the website and push to the `gh-pages` branch.
```
Comment (golota60) [inline www/README.md:41]: is this relevant for `react-bootstrap`?

**Review by golota60 — APPROVED** [review]
LGTM!
one question: do we wanna keep this section in the footer? doesn't seem to bring much value :smile: 


image

File: src/Pagination.tsx (hunk 1 above)
Comment (kyletsang) [reply src/Pagination.tsx:38]: I didn't think these were used (at least in the docs perspective), and not all our components with additional properties was documented like Modal or Form.  Is it needed?

File: www/README.md (hunk 2 above)
Comment (kyletsang) [reply www/README.md:41]: This was the default readme.  I'll clean this up

**Comment by kyletsang** [comment]
> LGTM! one question: do we wanna keep this section in the footer? doesn't seem to bring much value 😄
> 
> image

Footer looked better with 3 columns.  I can add more links

File: src/Pagination.tsx (hunk 1 above)
Comment (golota60) [reply src/Pagination.tsx:38]: i don't think they are - it's just jsdoc anyways :smile:

**Comment by jquense** [comment]
looks great. I wonder tho if its worth holding back merging to the real docs until we can add a RB theme to use? I feel really weird weird having a doc site for a UI library that doesn't use the UI library, dog fooding in this way is probably a minimum.

I also agree about the color, it's really bright and feels a little out of character, should we keep some of the older style greys?

**Comment by jquense** [comment]
I can probably take a stab at implementing the base theme btw, i've done it before, so should be relatively quick

**Comment by kyletsang** [comment]
> I also agree about the color, it's really bright and feels a little out of character, should we keep some of the older style greys?

I was thinking of using the older styles for dark mode on the site. Doesn't matter too much tho

**Comment by kyletsang** [comment]
Did you get a chance to look at the theme stuff @jquense?  Anything I can help with?

**Comment by jquense** [comment]
no :( i have no less than no time unfortunately. Lets not block on me since i have no idea when i could get to it now

**Comment by kyletsang** [comment]
No worries @jquense.  I'll merge this first since it's so large and we can iterate on it later.

**Pull request merged**